can/interfaces/socketcan/utils.py

```diff
@@ -68,24 +65,15 @@ def find_available_interfaces() -> Iterable[str]:
         return filter(_PATTERN_CAN_INTERFACE.match, interface_names)
 
 
-def error_code_to_str(code: int) -> str:
+def error_code_to_str(code: Optional[int]) -> str:
     """
     Converts a given error code (errno) to a useful and human readable string.
 
-    :param int code: a possibly invalid/unknown error code
-    :rtype: str
+    :param code: a possibly invalid/unknown error code
     :returns: a string explaining and containing the given error code, or a string
               explaining that the errorcode is unknown if that is the case
     """
+    name = errno.errorcode.get(code, "UNKNOWN") if code is not None else "UNKNOWN"
```

Comment:
Doesn't `get` already return the default when the key isn't in the dictionary? Something like:

```suggestion
    name = errno.errorcode.get(code, "UNKNOWN")
```

Reply:
Yep, that's right. But I added it for the type checker: It will complain when a `Mapping[int]` is queried with an `Optional[int]`. So either keep it as is or cast/ignore the type. And I thought this was the better approach. Probably should add a comment if that convinces you? ;)

Reply:
Nit: I asked over at python/typing, and it seems like adding `type: ignore` is the suggested way to go about this.

But why is `None` even allowed as input? Doesn't this function just take an `int` and consider it an error if anything else is passed to avoid this in the first place? The test `test_error_code_to_str` seems to assume this...

Reply:
I thought it might be useful where one is not sure whether an errno is present or not, though it is currently not used. Guess I'll add the `type: ignore` and change the test.